new-frontiers-14/frontier-station-14
# Pacified Zone: filter immune users by playtime

## About the PR



Adds an optional time requirement to the pacified zone.  This has been set to 10 hours for the default Frontier zones.

The common update logic in the PacifiedZoneSystem was pushed into a function to reduce redundancy.

## Why / Balance


On Frontier, the pacified zone's most useful on Frontier Station for preventing raiders, this makes it more in tune with that.

Changing the prototypes might be annoying for downstream forks - we could create ones specifically for the bus and Frontier that we use with these time limits.  Also, if this was to be used for anything else (e.g. NFSD riot beacons), this field would need to be removed.

## How to test


1. Spawn as a new character (connect as a new user to your local server).
2. Frontier's under pacified zone, the bus is under pacified zone (either check the dev map or wait for it to FTL)
3. Give your character 10 hours of playtime (`playtime_addoverall <localhost@JoeGenero> 36000`)
4. Respawn.  No pacified zone.

## Media



## Breaking changes


**Changelog**



:cl:
- tweak: Frontier's pacified zone now only affects users with 10 hours of playtime or less.
